configdialog.cpp

```diff
@@ -544,66 +544,259 @@ std::wstring PaletteCustomizeDialog::export_palettes() {
 		auto *data = palette.data;
 
 		palettes_string.append(palette.name + L": ");
-		palettes_string.append(m_palette_repo.serialize(*data));
-		palettes_string.append(L";\r\n");
+		std::wstring colors = m_palette_repo.serialize(*data) + L";";
+		// Add extra "fake" colors so that a palette definition is guaranteed to be longer than the maximum allowed name length
+		while (colors.length() < PaletteCustomizeDialog::MaxPaletteNameSize) {
+			colors.append(L"#000000;");
+		}
+		palettes_string.append(colors);
+		palettes_string.append(L"\r\n");
 	}
 	palettes_string.append(L"~");
 
 	return palettes_string;
 }
 
 void PaletteCustomizeDialog::import_palettes(std::wstring *palettes) {
-	static std::wstring valid_chars = L"qwertyuiopasdfghjklzxcvbnmQWERTYUIOPASDFGHJKLZXCVBNM0123456789-_!?:";
-	static std::wstring valid_hex_chars = L"0123456789abcdefABCDEF";
-	size_t current_index = palettes->find_first_of(valid_chars.c_str());
-	std::wstring last_successful_palette_name;
-	std::wstring current_name;
-	std::wstring current_data[7];
+	using namespace std;
+
+	const static wstring valid_chars = L"qwertyuiopasdfghjklzxcvbnmQWERTYUIOPASDFGHJKLZXCVBNM0123456789-_!?:";
+	const static wstring valid_hex_chars = L"0123456789abcdefABCDEF";
+	const static size_t colors_amount = max(7, (size_t)ceil(PaletteCustomizeDialog::MaxPaletteNameSize / 8.0));
+
+	size_t palettes_amount = 0;
+	size_t current_index = 0;
 	bool graceful_exit = false;
-	try {
-		while (true) {
-			current_name = palettes->substr(current_index, palettes->find_first_of(L':', current_index) - current_index);
-			if (current_name.find_first_not_of(valid_chars + L" ") != std::wstring::npos) {
-				throw ERROR_BAD_ARGUMENTS;
+	size_t count_invalid = 0;
+
+	const enum class Errors {
+		none = 0,
+		name_too_short = 1 << 0,
+		name_too_long = 1 << 1,
+		name_invalid_characters = 1 << 2,
+		colors_too_few = 1 << 3,
+		colors_incorrect_length = 1 << 4,
+		colors_invalid_characters = 1 << 5,
+	};
```

Comment:
This feels like an appropriate use of enums, no?

Reply:
It's debatable. It violates the "spirit" of an enum, which is that the variants are mutually-exclusive, but C++ enums are more or less just fancy names for integers and not much else is enforced, so eh, whatever.

However, given that 1) this type is not public-facing, so it does not help anyone else understand what the possible errors are and 2) you are `static_cast<uint_8>`ing like a motherfucker in order to treat these enum values like bit flags, I think you would be better off not using the enum here.